src/java.base/share/classes/java/io/DataInputStream.java

```diff
@@ -592,12 +597,15 @@ public static final String readUTF(DataInput in) throws IOException {
         int chararr_count=0;
 
         in.readFully(bytearr, 0, utflen);
+        int ascii = JLA.countPositives(bytearr, 0, utflen);
+        if (ascii == utflen) {
+            return new String(bytearr, 0, utflen, StandardCharsets.ISO_8859_1);
```

Comment:
Maybe use `sun.nio.cs.ISO_8859_1.INSTANCE` instead:
```suggestion
            return new String(bytearr, 0, utflen, ISO_8859_1.INSTANCE);
```

Reply:
We should use `StandardCharsets` as in most other standard usages. This method is not on bootstrap path, so readability is more important.